src/Microsoft.OData.Core/UriParser/Binders/InBinder.cs

```diff
@@ -92,6 +94,39 @@ private CollectionNode GetCollectionOperandFromToken(QueryToken queryToken, IEdm
                     replacedText[0] = '[';
                     replacedText[replacedText.Length - 1] = ']';
                     bracketLiteralText = replacedText.ToString();
+
+                    Debug.Assert(expectedType.IsCollection());
+                    if (expectedType.Definition.AsElementType().FullTypeName().Equals("Edm.String"))
+                    {
+                        // For collection of strings, need to convert single-quoted string to double-quoted string,
+                        // and also, per ABNF, two consecutive single quotes  to one single quote.
+                        // Sample: ['a''bc','''def','xyz'''] ==> ["a'bc","'def","xyz'"], which is legitimate Json format.
+                        string[] items = bracketLiteralText.Substring(1, bracketLiteralText.Length - 2).Split(',');
+
+                        // Skip conversion if the items are already in double-quote format (for backward compatibility).
+                        // Note that per ABNF, query option strings should use single quotes.
+                        if (items.Length > 0 && items[0][0] == '\'')

```

Comment:
Does the ABNF say that the use of single and double quotes have to be consistent? For example, is the client allowed to do `StringItem in ("Alan's computer", "Biao''s mouse, 'Sam''s keyboard')`?

Reply:
This is probably a nitpick comment and not a common scenario.

Reply:
Better be. :-)
Actually ABNF specifies using single-quote only in query options, per my understanding. But we allows double-quote now for collection used by IN operator, so we are trying to keep backward-compatibility.

Reply:
Makes sense. In that case, if we are checking `items[0][0] == '\''`, does that mean if I use `StringItem in ("Alan's computer", 'Sam''s keyboard')`, it will miss the second item in the list?

Reply:
Correct, it will miss the second item. This is only when inconsistent styles are used for items in the collection. :-) It would be an overkill to check for inconsistent styles?

Reply:
I guess since we are already diverging from the ABNF specification by allowing double-quote for backward-compatibility, it is up to us on how we want to treat inconsistencies. I think it's fine to just check based on the first item in that case.